lua/typst-preview/servers/manager.lua

```diff
@@ -32,10 +37,22 @@ function M.init(path, mode, callback)
     servers[path] == nil or servers[path][mode] == nil,
     'Server with path ' .. path .. ' and mode ' .. mode .. ' already exist.'
   )
-  factory.new(path, mode, function(server)
+
+  local port = config.opts.port
+  local link = '127.0.0.1:' .. tostring(port)
+  if links[link] ~= nil then
```

Comment:
I was more talking about when the port has already been used by another process. Not another typst-preview instance.